File: src/main/java/com/temelyan/pomoapp/util/UserUtil.java
package com.temelyan.pomoapp.util;

import com.temelyan.pomoapp.model.User;
import com.temelyan.pomoapp.to.UserTo;

public class UserUtil {
    public static UserTo asTo(User user) {
        return new UserTo(user.getId(), user.getEmail(), user.getPassword(), user.getProjects());
    }

    public static User updateFromTo(User user, UserTo userTo) {
        user.setEmail(userTo.getEmail().toLowerCase());
        user.setPassword(userTo.getPassword());
        return user;
    }

    public static User createNewFromTo(UserTo newUser) {
        return new User(null, newUser.getEmail().toLowerCase(), newUser.getPassword());
    }

    public static User prepareToSave(User user) {
        user.setPassword(PasswordUtil.encode(user.getPassword()));
        user.setEmail(user.getEmail().toLowerCase());
        return user;
    }
}

File: src/main/java/com/temelyan/pomoapp/util/UserUtil.kt
package com.temelyan.pomoapp.util

import com.temelyan.pomoapp.model.User
import com.temelyan.pomoapp.to.UserTo
import org.springframework.security.crypto.password.PasswordEncoder

object UserUtil {
    fun asTo(user: User): UserTo {
        return UserTo(user.getId(), user.email!!, user.password!!)
    }

    fun updateFromTo(user: User, userTo: UserTo): User {
        user.password = userTo.password
        return user
    }

    fun createNewFromTo(newUser: UserTo): User {
        return User(null, newUser.email!!.toLowerCase(), newUser.password!!)
    }

    fun prepareToSave(user: User, passwordEncoder: PasswordEncoder): User {
        user.password = PasswordUtil.encode(user.password!!, passwordEncoder)
        user.email = user.email!!.toLowerCase()
        return user
    }
}
